cpu/stm32_common/dist/clk_conf/clk_conf.c

```diff
@@ -0,0 +1,483 @@
+                }
+                return 1;
+            }
+        }
+        else {
+            puts("No other source available");
+            return 1;
+        }
+        break;
+    default:
+        break;
+    }
+
+    /* PLL I2S */
+    if (pll_i2s_p_out || pll_i2s_q_out) {
+        unsigned *_m;
+        unsigned _in;
+        if (cfg->has_pll_i2s_m) {
+            _m = &m_i2s;
+        }
+        else {
+            _m = &m;
+        }
+        if (cfg->has_pll_i2s_alt_input && pll_i2s_input) {
+            _in = pll_i2s_input;
+        }
+        else {
+            _in = pll_in;
+        }
+        if (compute_pll(&cfg->pll, _in, pll_i2s_p_out, pll_i2s_q_out, 0,
+                        _m, &n_i2s, &p_i2s, &q_i2s, &r_i2s) != 0) {
+            puts("Unable to compute 48MHz output using PLL I2S");
+            return 1;
+        }
+    }
+
+    /* PLL SAI */
+    if (pll_sai_p_out || pll_sai_q_out) {
+        if (compute_pll(&cfg->pll, pll_in, pll_sai_p_out, pll_sai_q_out, 0,
+                        &m_sai, &n_sai, &p_sai, &q_sai, &r_sai) != 0) {
+            puts("Unable to compute 48MHz output using PLL I2S");
+            return 1;
+        }
+        if (!cfg->has_pll_sai_m && m != m_sai) {
+            m = m_sai;
+            DEBUG("Retry to compute main PLL with M=%u\n", m);
+            if (compute_pll(&cfg->pll, pll_in, coreclock, clock_48MHz, 0,
+                            &m, &n, &p, &q, &r) < 0) {
+                puts("Unable to compute 48MHz output using PLL I2S");
+                return 1;
+            }
+        }
+    }
+
+    /* APB prescalers */
+    unsigned apb1_pre;
+    unsigned apb2_pre;
+
+    for (apb1_pre = 1; apb1_pre <= 16; apb1_pre <<= 1) {
+        if (coreclock / apb1_pre <= cfg->max_apb1) {
+            break;
+        }
+    }
+    for (apb2_pre = 1; apb2_pre <= 16; apb2_pre <<= 1) {
+        if (coreclock / apb2_pre <= cfg->max_apb2) {
+            break;
+        }
+    }
+
+
+    /* Print constants */
+    puts("==============================================================");
+    puts("Please copy the following code into your board's periph_conf.h");
+    puts("");
+
+    printf("/**\n"
+           " * @name Clock settings\n"
+           " *\n"
+           " * @note This is auto-generated from\n"
+           " *       `cpu/stm32_common/dist/clk_conf/clk_conf.c`\n"
+           " * @{\n"
+           " */\n");
+    printf("/* give the target core clock (HCLK) frequency [in Hz],\n"
+           " * maximum: %uMHz */\n", cfg->max_coreclock / 1000000U);
+    printf("#define CLOCK_CORECLOCK     (%uU)\n", coreclock);
+    printf("/* 0: no external high speed crystal available\n"
+           " * else: actual crystal frequency [in Hz] */\n"
+           "#define CLOCK_HSE           (%uU)\n", pll_src ? pll_in : 0);
+    printf("/* 0: no external low speed crystal available,\n"
+           " * 1: external crystal available (always 32.768kHz) */\n"
+           "#define CLOCK_LSE           (%d)\n", is_lse);
```

Comment:
how about `(%dU)` to make it consistent with `CLOCK_HSE`?

Reply:
Why not